include/swift/Basic/ParseableOutput.h

```diff
@@ -0,0 +1,72 @@
+//===----- ParseableOutput.h - Helpers for parseable output ------- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Helpers for emitting the compiler's parseable output.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_PARSEABLEOUTPUT_H
+#define SWIFT_PARSEABLEOUTPUT_H
+
+#include "swift/Basic/LLVM.h"
+#include "swift/Basic/TaskQueue.h"
+#include "swift/Driver/Job.h"
+#include "swift/Frontend/Frontend.h"
+
+namespace swift {
+
+namespace parseable_output {
+
+/// Quasi-PIDs are _negative_ PID-like unique keys used to
+/// masquerade batch job constituents as (quasi)processes, when writing
+/// parseable output to consumers that don't understand the idea of a batch
+/// job. They are negative in order to avoid possibly colliding with real
+/// PIDs (which are always positive). We start at -1000 here as a crude but
+/// harmless hedge against colliding with an errno value that might slip
+/// into the stream of real PIDs (say, due to a TaskQueue bug).
+const int QUASI_PID_START = -1000;
```

Comment:
Before I dig into the implementation details: When the driver assigns quasi-PIDs, it knows what all of the frontend jobs are doing and can prevent two of them from using the same quasi-PID. How are we preventing this if the frontend jobs themselves are assigning the quasi-PIDs?

(If you aren’t doing anything to handle this, one solution might be to use a different scheme. For instance, you could negate the real PID, multiply it by the batch size the driver was trying to assign to each job, and then add the index of the source file you’re currently building.)

Reply:
You're correct, there is currently nothing to coordinate this or avoid collisions among different frontend jobs, yet. I like your suggested scheme, let me think about this a little bit.